snowpack/src/plugins/plugin-esbuild.ts

```diff
@@ -29,12 +29,18 @@ export function esbuildPlugin(config: SnowpackConfig, {input}: {input: string[]}
     },
     async load({filePath}) {
       esbuildService = esbuildService || (await startService());
-      const contents = await fs.readFile(filePath, 'utf8');
+      let contents = await fs.readFile(filePath, 'utf8');
       const isPreact = checkIsPreact(filePath, contents);
+      const loader = getLoader(filePath);
       let jsxFactory = config.buildOptions.jsxFactory ?? (isPreact ? 'h' : undefined);
       let jsxFragment = config.buildOptions.jsxFragment ?? (isPreact ? 'Fragment' : undefined);
+
+      if (loader.endsWith('x') && config.buildOptions.jsxInject) {
+        contents = `${config.buildOptions.jsxInject}\n${contents}`;
```

Comment:
Does this become `"undefined\n${contents}"` if jsxInject is undefined? Also lets not add an extra `\n` if we can help it when `jsxInject` is not defined. What about instead:

```
const jsxInject = config.buildOptions.jsxInject ? `${config.buildOptions.jsxInject}\n` : '';
```

Reply:
If `jsxInject` is `undefined` this block should be skipped since `undefined` is falsy, but I do like your suggestion.